File: www/treemap-coloraxis/demo-grid-light.m
#import <UIKit/UIKit.h>
#import <Highcharts/Highcharts.h>

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    HIChartView *chartView = [[HIChartView alloc] initWithFrame:self.view.bounds];

    chartView.plugins = @[ @"heatmap", @"treemap" ];

    chartView.theme = @"grid-light";
    
    HIOptions *options = [[HIOptions alloc]init];
    
    HITreemap *treemap = [[HITreemap alloc]init];
    treemap.layoutAlgorithm = @"squarified";
    HIData *data1 = [[HIData alloc]init];
    data1.name = @"A";
    data1.value = @6;
    data1.colorValue = @1;
    HIData *data2 = [[HIData alloc]init];
    data2.name = @"B";
    data2.value = @6;
    data2.colorValue = @2;
    HIData *data3 = [[HIData alloc]init];
    data3.name = @"C";
    data3.value = @4;
    data3.colorValue = @3;
    HIData *data4 = [[HIData alloc]init];
    data4.name = @"D";
    data4.value = @3;
    data4.colorValue = @4;
    HIData *data5 = [[HIData alloc]init];
    data5.name = @"E";
    data5.value = @2;
    data5.colorValue = @5;
    HIData *data6 = [[HIData alloc]init];
    data6.name = @"F";
    data6.value = @2;
    data6.colorValue = @6;
    HIData *data7 = [[HIData alloc]init];
    data7.name = @"G";
    data7.value = @1;
    data7.colorValue = @7;
    treemap.data = @[data1, data2, data3, data4, data5, data6, data7];
    options.series = @[treemap];
    
    HITitle *title = [[HITitle alloc]init];
    title.text = @"Highcharts Treemap";
    options.title = title;
    
    options.additionalOptions = @{@"colorAxis": @{
                                          @"minColor": @"#FFFFFF",
                                          @"maxColor": @"#7cb5ec"
                                          }};
    
    chartView.options = options;
    
    [self.view addSubview:chartView];
}

@end

File: www-swift/euler-diagram/demo-grid-light.swift
import Highcharts
import UIKit

class ViewController: UIViewController {

  override func viewDidLoad() {
    super.viewDidLoad()

    let chartView = HIChartView(frame: view.bounds)
    chartView.theme = "grid-light"
    chartView.plugins = ["venn"]

    let options = HIOptions()

    let accessibility = HIAccessibility()
    accessibility.point = HIPoint()
    accessibility.point.valueDescriptionFormat = "{point.name}: {point.longDescription}."
    options.accessibility = accessibility

    let title = HITitle()
    title.text = "Relationship between Euler and Venn diagrams"
    options.title = title

    let tooltip = HITooltip()
    tooltip.headerFormat = "<span style=\"color:{point.color}\">\u{2022}</span><span style=\"font-size: 14px\"> {point.point.name}</span><br/>"
    tooltip.pointFormat = "{point.longDescription}<br><span style=\"font-size: 10px\">Source: Wikipedia</span>"
    options.tooltip = tooltip

    let venn = HIVenn()
    venn.name = "The Unattainable Triangle"

    let data1 = HIData()
    data1.sets = ["A"]
    data1.value = 4
    data1.name = "Euler diagrams"
    // data1.longDescription = "An Euler diagram is a diagrammatic means of representing sets and their relationships. Unlike Venn diagrams, which show all possible relations between different sets, the Euler diagram shows only relevant relationships."

    let data2 = HIData()
    data2.sets = ["B"]
    data2.value = 1
    data2.name = "Venn diagrams"
    // data2.longDescription = "In Venn diagrams the curves are overlapped in every possible way, showing all possible relations between the sets. They are thus a special case of Euler diagrams, which do not necessarily show all relations"

    let data3 = HIData()
    data3.sets = ["A", "B"]
    data3.value = 1

    venn.data = [data1, data2, data3]

    options.series = [venn]

    chartView.options = options

    self.view.addSubview(chartView)
  }

}
